app/src/androidTest/java/org/mozilla/fenix/syncintegration/SyncIntegrationTest.kt

```diff
@@ -109,7 +110,7 @@ class SyncIntegrationTest {
     }
 
     fun tapOnSignIn() {
-        mDevice.wait(Until.findObjects(By.text("Sign in")), TestAssetHelper.waitingTime)
+        assertNotNull(mDevice.wait(Until.findObjects(By.text("Sign in")), TestAssetHelper.waitingTime))
```

Comment:
It occurs to me that we could make this easier to reuse with an extension function.  Something like the following:

```kotlin
    /**
     * Blocks the test for [waitTime] miliseconds before continuing.
     *
     * Will cause the test to fail is the condition is not met before the timeout.
     */
    fun UiDevice.waitNotNull(
        searchCondition: SearchCondition<*>,
        waitTime: Long = TestAssetHelper.waitingTimeShort
    ) = assertNotNull(wait(searchCondition, waitTime))
```

Which would update L113 to look like `mDevice.waitNotNull(Until.findObjects(By.text("Sign in")))`, or (if we wanted to use a different timeout) `mDevice.waitNotNull(Until.findObjects(By.text("Sign in")), TestAssetHelper.waitingTimeShort)`

Reply:
^ @rpappalax fyi

Reply:
We can open a new issue.